src/demotivation.py

```diff
@@ -0,0 +1,94 @@
+from diffusers import StableDiffusionPipeline
+from transformers import FSMTForConditionalGeneration, FSMTTokenizer, VisionEncoderDecoderModel, ViTFeatureExtractor, \
+    AutoTokenizer, pipeline
+import torch
+from simpledemotivators import Demotivator
+import streamlit as st
+from PIL import Image
+
+ENG_TO_RU_MODEL = "facebook/wmt19-en-ru"
+IMAGE_MODEL = "DGSpitzer/Cyberpunk-Anime-Diffusion"
+IMAGE_CAPTION_MODEL = "nlpconnect/vit-gpt2-image-captioning"
+DETECTION_LANGUAGE_MODEL = "papluca/xlm-roberta-base-language-detection"
+DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+@st.cache(allow_output_mutation=True, suppress_st_warning=True)
+def get_pipe_image():
+    if DEVICE == "cuda":
+        return StableDiffusionPipeline.from_pretrained(IMAGE_MODEL, torch_dtype=torch.float16, revision="fp16",
+                                                       requires_safety_checker=False).to(DEVICE)
+    else:
+        return StableDiffusionPipeline.from_pretrained(IMAGE_MODEL, requires_safety_checker=False).to(DEVICE)
+
+
+def generate_image(text, path, steps, height, width):
+    pipe_image = get_pipe_image()
+    pipe_image.safety_checker = lambda images, clip_input: (images, False)
+    image = pipe_image(text, num_inference_steps=steps, height=height, width=width).images[0]
+    image.save(path)
```

Comment:
Is it possible to avoid image saving?

Reply:
Removed saving the image from the demotivator, left saving the image to the buffer in streamlit for the download button, since the button requires either bytes or the path to the saved image.